feat: allow definition of custom metrics with filters (#5865)
lightdash/lightdash: docs/docs/snippets/how-to-create-metrics.mdx

So, we're going to break it down for you: what are custom metrics? and how can I use them? 🪄

### Creating a custom metric from a dimension

To create a custom metric from a dimension, you just need to:

1. Click on the dimension's three-dot `options` menu
2. Click on one of the options available (e.g. `Count Distinct`)
3. Confirm your changes

<img src={CustomMetricCreating} width="300" height="326" style={{display: "block", margin: "0 auto 20px auto"}}/>

Then, your new metric will be added to your results table automatically and will appear in the `custom metrics` space in your sidebar.

<img src={CustomMetricAdded} width="928" height="690" style={{display: "block", margin: "0 auto 20px auto"}}/>

If you want to delete the custom metric, you can just click on the three-dot `options` menu and hit `Remove custom metric`

<img src={CustomMetricDeleting} width="300" height="137" style={{display: "block", margin: "0 auto 20px auto"}}/>

### Setting filters to your custom metric

These filters are based on the dimensions set in your project, allowing for more precise metrics.

> The ability to add filters is now available when you're creating a new custom metric or editing an existing one.

Reviewer: ```suggestion
You can add filters when you're creating a new custom metric or editing an existing one.
```

Author: thanks!
